setup.sh

```diff
@@ -1,66 +1,102 @@
-OS_IS_SUPPORTED=0
-DOCKER_COMPOSE_BIN=/usr/local/bin/docker-compose
+# Variables
+KUZZLE_DIR="./kuzzle"
+KUZZLE_DOWNLOAD_MANAGER=""
+KUZZLE_PUSH_ANALYTICS=""
+KUZZLE_CHECK_DOCKER_COMPOSE_YML_HTTP_STATUS_CODE=""
+KUZZLE_CHECK_CONNECTIVITY_CMD=""
+CURL_OPTS="-sSL"
+CURL_PUSH_OPTS="-H Content-Type:application/json --data "
+WGET_OPTS="-qO-"
+WGET_PUSH_OPTS=" -O- --header=Content-Type:application/json --post-data="
 ANALYTICS_URL="http://analytics.kuzzle.io/"
 GITTER_URL="https://gitter.im/kuzzleio/kuzzle"
-COMPOSER_YML_URL="https://raw.githubusercontent.com/kuzzleio/kuzzle-build/master/docker-compose/kuzzle-docker-compose.yml"
 SUPPORT_MAIL="support@kuzzle.io"
-KUZZLE_DIR="${HOME}/.kuzzle"
-COMPOSER_YML_PATH="${KUZZLE_DIR}/docker-compose.yml"
-FIRST_INSTALL=0
+COMPOSE_YML_URL="https://kuzzle.io/docker-compose.yml"
+COMPOSE_YML_PATH=$KUZZLE_DIR/docker-compose.yml
+INSTALL_KUZZLE_WITHOUT_DOCKER_URL="https://docs.kuzzle.io/guide/essentials/installing-kuzzle/#manually"
+MIN_DOCKER_VER=1.12.0
+MIN_MAX_MAP_COUNT=262144
+CONNECT_TO_KUZZLE_MAX_RETRY=30
+CONNECT_TO_KUZZLE_WAIT_TIME_BETWEEN_RETRY=2 # in seconds
+DOWNLOAD_DOCKER_COMPOSE_YML_MAX_RETRY=3
+DOWNLOAD_DOCKER_COMPOSE_RETRY_WAIT_TIME=1 # in seconds
+OS=""
+
+# Errors return status
+NO_INTERNET=42
+NO_DOWNLOAD_MANAGER=43
+NO_DOCKER=44
+NO_DOCKER_COMPOSE=45
+DOCKER_VERSION_MISMATCH=46
+MIN_MAX_MAP_COUNT_MISMATCH=47
+ERROR_DOWNLOAD_DOCKER_COMPOSE=48
+KUZZLE_NOT_RUNNING_AFTER_INSTALL=49
 
-if [ ! -d $KUZZLE_DIR ]; then
-  mkdir $KUZZLE_DIR;
-fi
 
-if [ ! -f "${KUZZLE_DIR}/uid" ]; then
-  echo $(LC_CTYPE=C tr -dc A-Fa-f0-9 < /dev/urandom | fold -w ${1:-64} | head -n 1) > "${KUZZLE_DIR}/uid"
-  FIRST_INSTALL=1
+# list of colors
+# see if it supports colors...
+NCOLORS=$(tput colors)
+if [ "$NCOLORS" -gt 0 ]; then
+  BOLD=$(tput bold)
+  RED=$(tput setaf 1)
+  BLUE=$(tput setaf 4)
+  NORMAL=$(tput sgr0)
+  GREEN="$(tput setaf 2)"  
 fi
 
-UUID=$(cat "${KUZZLE_DIR}/uid")
-
-trap abortSetup INT
-
-function abortSetup() {
-  if [ ! -z $DL_BIN ]; then
-    $DL_BIN $UL_OPTS '{"type": "abort-setup", "uid": "'$UUID'"}' $ANALYTICS_URL &> /dev/null
-  fi
-  exit 1
-}
-
-# Output a text with the selected color (reinit to normal at the end)
-write() {
-  echo -e " $1$2" "$NORMAL"
-}
+# Create kuzzle workspace directory
+if [ ! -d $KUZZLE_DIR ]; then
+  mkdir $KUZZLE_DIR;
+fi
 
-writeBold() {
-  echo -e "${BOLD} $1$2" "$NORMAL"
+os_lookup() {
+  OSTYPE=$(uname)
+  case "$OSTYPE" in
+    "Darwin")
+    {
+        OS="OSX"
+    } ;;    
+    "Linux")
+    {
+        # If available, use LSB to identify distribution
+        if [ -f /etc/lsb-release -o -d /etc/lsb-release.d ]; then
+            DISTRO=$(awk -F= '/^NAME/{print $2}' /etc/os-release)
+        else
+            DISTRO=$(ls -d /etc/[A-Za-z]*[_-][rv]e[lr]* | grep -v "lsb" | cut -d'/' -f3 | cut -d'-' -f1 | cut -d'_' -f1)
+        fi
+        OS=$(echo $DISTRO | tr 'a-z' 'A-Z' | tr -d '"')
+    } ;;
+    *) 
+    {
+        OS=OSTYPE
+        exit
```

Comment:
if the OS is neither "darwin" nor "linux", the script exits silently, without any message or information about what's going on

what if someone tries on a non-linux, posix2 OS supported by docker?